### PR TITLE
Add initial implementation and corresponding variables

### PR DESCRIPTION
**Pull request by umairidris**

### CODE REVIEW

**Comment by umairidris** [comment]
cc @morgante

File: main.tf [hunk 1]
```diff
@@ -14,11 +14,114 @@
  * limitations under the License.
  */
 
-terraform {
-  required_version = "~> 0.12.0"
+resource "google_healthcare_dataset" "dataset" {
+    provider = google-beta
```
Comment (morgante) [inline main.tf]: Please run `terraform fmt`.

File: main.tf [hunk 2]
```diff
@@ -14,11 +14,114 @@
...
+            for m in s.iam_members: {
+                store_name = s.name
+                role = m.role
+                member = m.member
+            }
+        ]
+    ])
+}
+
+resource "google_healthcare_dicom_store_iam_member" "dicom_store_iam_members" {
```
Comment (morgante) [inline main.tf]: I'd suggest we split all iam resources into an `iam.tf` file.

File: variables.tf [hunk 3]
```diff
@@ -14,10 +14,69 @@
...
+variable "name" {
+    type = string
+    description = "The resource name for the Dataset.s"
 }
 
-variable "bucket_name" {
-  description = "The name of the bucket to create"
+variable "project" {
+    type = string
+    description = "The ID of the project in which the resource belongs. If it is not provided, the provider project is used."
```
Comment (morgante) [inline variables.tf]: We should mandate explicitly setting a project, since that's what we usually do on CFT.

File: main.tf (hunk 1 above)
Comment (umairidris) [reply main.tf]: Done.

File: main.tf (hunk 2 above)
Comment (umairidris) [reply main.tf]: Done.

File: variables.tf (hunk 3 above)
Comment (umairidris) [reply variables.tf]: Done.

**Comment by umairidris** [comment]
Thanks for the review! Addressed the comments. I will update the README and add tests, etc in a follow up!

File: iam.tf [hunk 4]
```diff
@@ -0,0 +1,73 @@
+locals {
```
[the remark this answers is not in the document]
Comment (umairidris) [reply iam.tf]: Done.

**Pull request merged**